source/Hazard.cpp

```diff
@@ -0,0 +1,120 @@
+/* Hazard.cpp
+Copyright (c) 2020 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Hazard.h"
+
+#include "DataNode.h"
+#include "Effect.h"
+#include "GameData.h"
+#include "Random.h"
+
+using namespace std;
+
+
+void Hazard::Load(const DataNode &node)
+{
+	if(node.Size() < 2)
+		return;
+	name = node.Token(1);
+	
+	for(const DataNode &child : node)
+	{
+		const string &key = child.Token(0);
+		if(key == "weapon")
+			LoadWeapon(child);
+		else if(key == "no deviation")
+			deviates = false;
+		else if(child.Size() < 2)
+			child.PrintTrace("Skipping hazard attribute with no value specified:");
+		else if(key == "period")
+			period = max(1, static_cast<int>(child.Value(1)));
+		else if(key == "length")
+		{
+			minLength = max(0, static_cast<int>(child.Value(1)));
+			maxLength = max(0, (child.Size() >= 3 ? static_cast<int>(child.Value(2)) : 0));
+		}
+		else if (key == "strength")
+		{
+			minStrength = max(0., child.Value(1));
+			maxStrength = max(0., (child.Size() >= 3) ? child.Value(2) : 0.);
+		}
+		else if(key == "range")
+		{
+			minRange = max(0., (child.Size() >= 3) ? child.Value(1) : 0.);
+			maxRange = max(0., (child.Size() >= 3) ? child.Value(2) : child.Value(1));
+		}
+		else if(key == "environmental effect")
+		{
+			int count = (child.Size() >= 3) ? child.Value(2) : 1;
```

Comment:
This is copied from a place like Weapon.cpp, where no such check for negative effect counts is done. If we do something here, we should reevaluate all other effects as well. I imagine though that a negative value is treated like zero, because somewhere along the line this count is just being fed into a for loop with the counter initialized at 0.

Reply:
We can evaluate other effects in a different PR